tests/framework/extensions/configmaps/configmaps.go

```diff
@@ -3,3 +3,19 @@ package configmaps
 const (
 	ConfigMapSteveType = "configmap"
 )
+
+type GenFieldTest struct {
```

Comment:
Because I'm sending one with fields that I didn't think were available with that struct:
```go
c.configMapPayload.Metadata.Fields = "test-fields"
c.configMapPayload.Metadata.Relationships = "test-relationships"
c.configMapPayload.Metadata.State = "test-state"
```
Know of an example for that with your link?

Reply:
So this a different struct? How is it related to configmaps? That's the part I'm confused about, if you don't mind explaining

Reply:
Yeah the struct has additional fields that are added on when using the UI. So, it's mimicking the payload if you did this action via the UI. These ones:

- metadata.fields
- metadata.relationships
- metadata.state

Reply:
I understand so then you should use the Steve object, and update the metadata include fields and relationships as I believe those are new things. Here: https://github.com/rancher/rancher/blob/7d70ef85a98e7adbb38c917905cae4795d750249/tests/framework/clients/rancher/v1/client.go#L34

Reply:
Updated this, added these fields in. Still using a custom struct in configmaps, but using Object meta, and type meta after huddling with @igomez06 so it's closer to test case.